src/App.tsx

```diff
@@ -1,61 +1,149 @@
+import { useState, FormEvent } from 'react';
 import './App.scss';
+import usersFromServer from './api/users';
+import todosFromServer from './api/todos';
+import { TodoList } from './components/TodoList';
+import { PreparedTodo } from './types/PreparedTodo';
 
-// import usersFromServer from './api/users';
-// import todosFromServer from './api/todos';
+const getNewTodoId = (todoList: PreparedTodo[]) => {
+  const usedIds = todoList.map(todo => todo.id).sort((a, b) => a - b);
+  const freeId = usedIds.findIndex((usId, index) => usId !== index + 1);
+  const newId = freeId !== -1 ? freeId + 1 : usedIds.length + 1;
```

Comment:
![image](https://user-images.githubusercontent.com/91411692/234606693-732c5a15-853a-4ff7-80eb-4c0315ce4b08.png)

```suggestion
  const largestId = usedIds[usedIds.length - 1].id;
  const newId = largestId + 1;
```